src/components/modals/ModalContext.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { Reducers } from 'openshift-assisted-ui-lib';
+
+const {
+  dialogsReducer,
+  openDialog: openDialogAction,
+  closeDialog: closeDialogAction,
+} = Reducers.Dialogs;
+
+type DialogId = 'downloadIsoDialog'; // TODO: list additional dialogs here
+
+type DownloadIsoDialogProps = {
+  fileName: string;
+  downloadUrl: string;
+};
+
+type ModalDialogsDataTypes = {
+  downloadIsoDialog: DownloadIsoDialogProps;
+};
+
+export type ModalDialogsContextType = {
+  [key in DialogId]: {
+    isOpen: boolean;
+    open: (data: ModalDialogsDataTypes[key]) => void;
+    onClose: () => void;
+    data?: ModalDialogsDataTypes[key];
+  };
+};
+
+const dialogIds: DialogId[] = ['downloadIsoDialog'];
+
+const ModalDialogsContext = React.createContext<ModalDialogsContextType | undefined>(undefined);
+
+// TODO(mlibra): reuse ModalDialogsContextProvider from the assisted-ui-lib
```

Comment:
Should we use console's own modal mechanisms to handle the dialog?

Reply:
Ok, let's keep that mechanism on the "consumer side" and do not expose it via API for now.

My thinking is, we do not need to be bound to the OCP Console mechanism, it's dynamic-cim local decision.

IMO, we can keep that mechanism here as a `dynamic-cim` implementation detail. However, I do not have strong opinion on keeping it at all, we can simplify to avoid using of the context.